src/cloudsc_loki/cloudsc_driver_field_loki_mod.F90

```diff
@@ -0,0 +1,140 @@
+    CALL TIMER%START(NUMOMP)
+
+    ! Workaround for PGI / OpenACC oddities:
+    ! Create a local copy of the parameter struct to ensure they get
+    ! moved to the device the in ``acc data`` clause below
+    LOCAL_YRECLDP = YRECLDP
+
+    !$loki data
+
+    !$omp parallel default(shared) private(JKGLO,IBL,ICEND,TID) &
+    !$omp& num_threads(NUMOMP) firstprivate(PAUX, FLUX, TENDENCY_TMP, TENDENCY_LOC)
+
+    ! Local timer for each thread
+    TID = GET_THREAD_NUM()
+    CALL TIMER%THREAD_START(TID)
+
+    !$omp do schedule(runtime) reduction(+:power_total,power_count)
+    DO JKGLO=1,NGPTOT,NPROMA
+        IBL=(JKGLO-1)/NPROMA+1
+        ICEND=MIN(NPROMA,NGPTOT-JKGLO+1)
+
+        CALL PAUX%UPDATE_VIEW(IBL)
+        CALL FLUX%UPDATE_VIEW(IBL)
+        CALL TENDENCY_LOC%UPDATE_VIEW(IBL)
+        CALL TENDENCY_TMP%UPDATE_VIEW(IBL)
+
+        !-- These were uninitialized : meaningful only when we compare error differences
+        PAUX%PCOVPTOT = 0.0_JPRB
+        TENDENCY_LOC%CLD(:,:,NCLV) = 0.0_JPRB
+
+
+
+        CALL CLOUDSC(    1,    ICEND,    NPROMA,  NLEV, & ! These could also be accessed through FIELD_STATE
+              & PTSPHY,&
+              & PAUX%PT, PAUX%PQ, &
+              & TENDENCY_TMP%T, TENDENCY_TMP%Q, TENDENCY_TMP%A, TENDENCY_TMP%CLD, &
+              & TENDENCY_LOC%T, TENDENCY_LOC%Q, TENDENCY_LOC%A, TENDENCY_LOC%CLD, &
+              & PAUX%PVFA, PAUX%PVFL, PAUX%PVFI, PAUX%PDYNA, PAUX%PDYNL, PAUX%PDYNI, &
+              & PAUX%PHRSW,    PAUX%PHRLW,&
+              & PAUX%PVERVEL,  PAUX%PAP,      PAUX%PAPH,&
+              & PAUX%PLSM,     PAUX%LDCUM,    PAUX%KTYPE, &
+              & PAUX%PLU,      PAUX%PLUDE,    PAUX%PSNDE,    PAUX%PMFU,     PAUX%PMFD,&
+              !---prognostic fields
+              & PAUX%PA,&
+              & PAUX%PCLV,  &
+              & PAUX%PSUPSAT,&
+!             -- arrays for aerosol-cloud interactions
+!             !! & PQAER,    KAER, &
+              & PAUX%PLCRIT_AER,PAUX%PICRIT_AER,&
+              & PAUX%PRE_ICE,&
+              & PAUX%PCCN,     PAUX%PNICE,&
+              !---diagnostic output
+              & PAUX%PCOVPTOT, PAUX%PRAINFRAC_TOPRFZ,&
+              !---resulting fluxes
+              & FLUX%PFSQLF,   FLUX%PFSQIF ,  FLUX%PFCQNNG,  FLUX%PFCQLNG,&
+              & FLUX%PFSQRF,   FLUX%PFSQSF ,  FLUX%PFCQRNG,  FLUX%PFCQSNG,&
+              & FLUX%PFSQLTUR, FLUX%PFSQITUR , &
+              & FLUX%PFPLSL,   FLUX%PFPLSN,   FLUX%PFHPSL,   FLUX%PFHPSN, &
+              & LOCAL_YRECLDP)
+
+#ifndef CLOUDSC_GPU_TIMING
+        ! Log number of columns processed by this thread
+        CALL TIMER%THREAD_LOG(TID, IGPC=ICEND)
+#endif
+      ENDDO
+
+      !-- The "nowait" is here to get correct local timings (tloc) per thread
+      !   i.e. we should not wait for slowest thread to finish before measuring tloc
+      !$omp end do nowait
+
+      CALL TIMER%THREAD_END(TID)
+
+      !$omp end parallel
+
+      !$loki end data
+
+      CALL TIMER%END()
+
```

Comment:
Please add here the following block 
```suggestion

#ifdef CLOUDSC_GPU_TIMING
    ! On GPUs, adding block-level column totals is cumbersome and
    ! error prone, and of little value due to the large number of
    ! processing "thread teams". Instead we register the total here.
    CALL TIMER % THREAD_LOG(TID=TID, IGPC=NGPTOT)
#endif
```
Otherwise the throughput performance reporting is broken when running on GPU

as in the other loki_driver_mod https://github.com/ecmwf-ifs/dwarf-p-cloudsc/blob/82fdf4b4021b37962c3c129c7b0047d72c39ee20/src/cloudsc_loki/cloudsc_driver_loki_mod.F90#L182-L187